File: 771. Jewels and Stones/_771.py
class Solution:
    def numJewelsInStones(self, J: str, S: str) -> int:
        res = 0
        for i in S:
            if i in J:
                res += 1
        return res

    def numJewelsInStones1(self, J, S):
        return sum(map(J.count, S))
    
    def numJewelsInStones2(self, J, S):
        return sum(map(S.count, J)) 

    def numJewelsInStones3(self, J, S):             
        return sum(s in J for s in S)
        #  return sum([i in J for i in S])

File: 771. Jewels and Stones/771.go
package leetcode

func numJewelsInStones(J string, S string) int {
	myDict := make(map[rune]int, len(J))
	for _, v := range J{
		myDict[v] = 1
	}
	res := 0
	for _, v := range S{
		_, ok := myDict[v]
		if ok{
			res ++
		}
	}
	return res
}

// 这有点时间换空间的感觉，感觉和上面的方法差不太多，主要就少了一个map
func numJewelsInStones1(J string, S string) int {
    count := 0
    in := func(slice string, s string) bool {
        for _, el := range slice {
            if string(el) == s {
                return true
            }
        }
        return false
    }
    for _, el := range S {
        if in(J, string(el)){
            count++
        }
    }
    return count
}